include/seqan3/search/algorithm/detail/search.hpp

```diff
@@ -180,36 +181,32 @@ inline auto search_single(index_t const & index, query_t & query, configuration_
 {
-    using cfg_t = remove_cvref_t<configuration_t>;
-    // return type: for each query: a vector of text_positions (or cursors)
-    // delegate params: text_position (or cursor). we will withhold all hits of one query anyway to filter
-    //                  duplicates. more efficient to call delegate once with one vector instead of calling
-    //                  delegate for each hit separately at once.
-    using text_pos_t = std::conditional_t<index_t::text_layout_mode == text_layout::collection,
-                                          std::pair<typename index_t::size_type, typename index_t::size_type>,
-                                          typename index_t::size_type>;
-    using hit_t = std::conditional_t<cfg_t::template exists<search_cfg::output<detail::search_output_index_cursor>>(),
-                                     typename index_t::cursor_type,
-                                     text_pos_t>;
-
-    if constexpr (std::ranges::forward_range<queries_t> && std::ranges::random_access_range<value_type_t<queries_t>>)
+    if constexpr (!std::ranges::forward_range<std::ranges::range_value_t<queries_t>>)
     {
-        // TODO: if constexpr (contains<search_cfg::id::on_hit>(cfg))
-        std::vector<std::vector<hit_t>> hits;
-        hits.reserve(std::distance(queries.begin(), queries.end()));
-        for (auto const query : queries)
-        {
-            hits.push_back(search_single(index, query, cfg));
-        }
-        return hits;
+        return search_all(index, std::views::single(std::forward<queries_t>(queries)), cfg);
     }
-    else // std::ranges::random_access_range<queries_t>
+    else
     {
-        // TODO: if constexpr (contains<search_cfg::id::on_hit>(cfg))
-        return search_single(index, queries, cfg);
+        using search_traits_t = search_traits<configuration_t>;
+
+        // return type: for each query: a vector of text_positions (or cursors)
+        // delegate params: text_position (or cursor). we will withhold all hits of one query anyway to filter
+        //                  duplicates. more efficient to call delegate once with one vector instead of calling
+        //                  delegate for each hit separately at once.
+        using text_pos_t = std::conditional_t<index_t::text_layout_mode == text_layout::collection,
+                                              std::pair<typename index_t::size_type, typename index_t::size_type>,
+                                              typename index_t::size_type>;
+        using hit_t = std::conditional_t<search_traits_t::search_return_index_cursor,
+                                         typename index_t::cursor_type,
+                                         text_pos_t>;
+        using query_t = std::ranges::range_value_t<queries_t>;
+        using search_fn_t = std::function<std::vector<hit_t>(query_t &)>;
+
+        search_fn_t search_fn = [&, cfg] (query_t & query) { return search_single(index, query, cfg); };
```

Comment:
not sure. Was from 3 months back. Can't remember. Will remove.

Reply:
Ah I remember why. The config might have been changed within the call to search. But search returns another object which captures an invocable. This lambda. When I return that range the config created within the call of search goes out of scope and finally we have a dangling reference. So this must be copied here to make sure it can be used safely.